src/map/map.ts

```diff
@@ -603,12 +604,25 @@ export class PropertiesMap {
         };
 
         // ======= markers size
-        const sizeChange = () => {
+
+        this._options.size.factor.onchange = () => {
             this._restyle({ 'marker.size': this._sizes(0) } as Data, 0);
         };
 
-        this._options.size.property.onchange = sizeChange;
-        this._options.size.factor.onchange = sizeChange;
+        this._options.size.mode.onchange = () => {
+            if (this._options.size.mode.value !== 'constant') {
+              this._options.size.property.enable();
+              // hacky default to first plotted parameter
+              this._options.size.property.value = this._options.x.property.value;
```

Comment:
Why do we need to change `this._options.size.property.value`? Could we just keep it as when enabling/disabling? 

And since `this._options.size.property.value = ''` is no longer a valid value, how about changing the validation & initial value directly? Using the same property as x axis works for me!

Reply:
As a user, I'd want to see some immediate feedback on the plot when switching size modes -- this won't happen if we don't designate some property. _However_, I should write it only to set as x axis if we're going from constant to not constant. Will patch this morning.

Reply:
> As a user, I'd want to see some immediate feedback on the plot when switching size modes -- this won't happen if we don't designate some property.

Completely agree with you. My point was more about removing the possibility of property being and empty string, by changing the validator & initial value in MapOptions, removing the need for changing it here

Reply:
So I just pushed a few changes. One non-trivial one is I added an enabled() and disabled() function to the HTMLOptions to check if an element is enabled. That way, for this and as well as others, we don't have to rely on values being empty strings to check if something is enabled.

Reply:
👍